components/public-api-server/pkg/apiv1/workspace.go

```diff
@@ -76,6 +76,50 @@ func (s *WorkspaceService) GetWorkspace(ctx context.Context, req *connect.Reques
 	}), nil
 }
 
+func (s *WorkspaceService) WorkspaceStatusUpdate(ctx context.Context, req *connect.Request[v1.WorkspaceStatusUpdateRequest], stream *connect.ServerStream[v1.WorkspaceStatusUpdateResponse]) error {
+	logger := ctxlogrus.Extract(ctx)
+
+	conn, err := getConnection(ctx, s.connectionPool)
+	if err != nil {
+		return err
+	}
+
+	workspace, err := conn.GetWorkspace(ctx, req.Msg.GetWorkspaceId())
+	if err != nil {
+		logger.WithError(err).Error("Failed to get workspace.")
+		return proxy.ConvertError(err)
+	}
+
+	if workspace.LatestInstance == nil {
+		logger.WithError(err).Error("Failed to get latest instance.")
+		return connect.NewError(connect.CodeFailedPrecondition, fmt.Errorf("instance not found"))
+	}
+
+	ch, err := conn.InstanceUpdates(ctx, workspace.LatestInstance.ID)
+	if err != nil {
+		logger.WithError(err).Error("Failed to get workspace instance updates.")
+		return proxy.ConvertError(err)
+	}
+
+	for {
+		select {
+		case update := <-ch:
+			instance, err := convertWorkspaceInstance(update, workspace.Workspace.Shareable)
+			if err != nil {
+				logger.WithError(err).Error("Failed to convert workspace instance.")
+				return proxy.ConvertError(err)
+			}
+			_ = stream.Send(&v1.WorkspaceStatusUpdateResponse{
+				Result: &v1.WorkspaceStatus{
+					Instance: instance,
+				},
+			})
```

Comment:
You'll need to handle the case when `ch` closes. `ch` will close when server restarts or otherwise the connection is terminated.